pkg/actors/actors.go

```diff
@@ -182,6 +194,7 @@ func NewActors(
 		resiliency:             resiliency,
 		storeName:              stateStoreName,
 		isResiliencyEnabled:    configuration.IsFeatureEnabled(features, configuration.Resiliency),
+		internalActors:         internalActors,
```

Comment:
Is there a way to avoid separating internal vs non-internal actors?

Reply:
I think the alternative would be to use regular actors. However, this seemed like a bad idea from a performance perspective since it introduces an unnecesary gRPC network hop into the same process for local actors. It also complicates the app channel abstraction because we need to support two types of application endpoints, one for internal endpoints (inside the sidecar) and another for endpoints defined in the app. The current design to distinguish internal vs. external actors felt more natural to me and less obtrusive to the overall Dapr sidecar architecture.

Reply:
My point was not to avoid the separation as concepts but to wrap them under the same interface, so we can abstract the fact that one is over gRPC and another one is over regular method invocation.

Reply:
I think the best we can do is create a new `AppChannel` implementation for internal actors. Currently there's a single "global" app channel for invoking the app over gRPC. We could create a second one for making local calls and no-op things like health checks. Would that be better?